### ISSUE TITLE
Fuzzy matcher should favour whole words

### ISSUE THREAD

**Issue by davep** [Textualize/textual#3208]
*[Meta: note that changes to the fuzzy matcher are in process in #3058 -- this is mostly a reminder to myself to either make this change there are do this as a close followup once the PR is merged]*

The fuzzy matcher should favour finding a complete word over finding the start of the letter collection as early as possible. The issue can be seen here:

<img width="169" alt="Screenshot 2023-08-30 at 14 52 55" src="https://github.com/Textualize/textual/assets/28237/62b6eb10-b01c-45a8-a1a9-f08325e03cc2">

`poo` should match `poo` in `pile_of_poo` (and so giving a higher score) in preference to matching the earlier instances of `p` and `o`.

**Comment by davep** [comment]
@gunslingerfry Note that this is quite an old issue, and also I've not been involved with Textual for quite some time now. I think I saw mention that the maintainer made some changes to the default matcher; it might be sensible for you to raise an issue if you feel you've found a problem.

**Comment by gunslingerfry** [comment]
My apologies. I will create a new issue. Thanks!
